### ISSUE TITLE
Support ability to simulate delayed channel connection completion

### ISSUE THREAD

**Issue by emerkle826** [datastax/simulacron#95]
Currently, there are 3 modes for [RejectScope](https://github.com/datastax/simulacron/blob/master/native-server/src/main/java/com/datastax/oss/simulacron/server/RejectScope.java#L19-L22) that allow Simulacron to simulate connection issues: UNBIND, STOP and REJECT_STARTUP. Unfortunately, none of these is able to simulate a situation where the server takes a while to complete a channel connection attempt from a client. It would be nice to have this ability as it would allow for testing client channel connection timeouts.

**Comment by jorgebay** [comment]
Good idea, it would help us testing the drivers against this scenario easily.
